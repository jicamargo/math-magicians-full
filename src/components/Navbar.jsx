import { useRef } from "react";
import { NavLink } from "react-router-dom";
import React from 'react';
import PropTypes from "prop-types";

const links = [
  { path: 'Home', text: 'Home' },
  { path: 'Calc', text: 'Calculator' },
  { path: 'Quote', text: 'Quote' },
];

const Navbar = () => {

  const ref = useRef();

  return (
    <>
      <nav ref={ref} className="navbar">
        <h1 className="menu-icon">Math Magicians</h1>
        <ul className={"menu-nav"}>
          {links.map((link) => {
            return (
              <React.Fragment key={link.text}>
                    <li>
                      <NavLink to={link.path}>
                        {link.text}
                      </NavLink>
                    </li>
              </React.Fragment>
            );
          })}
        </ul>        
      </nav>
    </>
  );
};

Navbar.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
    })
  ),
};

export default Navbar;