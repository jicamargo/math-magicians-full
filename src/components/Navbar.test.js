// eslint-disable-next-line no-unused-vars
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('renders the Navbar component', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    expect(screen.getByRole('navigation')).toMatchSnapshot();
  });

  it('renders the navigation links correctly', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    const navLinks = screen.getAllByRole('link');

    const expectedLinks = [
      { path: 'Home', text: 'Home' },
      { path: 'Calc', text: 'Calculator' },
      { path: 'Quote', text: 'Quote' },
    ];

    navLinks.forEach((navLink, index) => {
      expect(navLink.getAttribute('href')).toEqual(`/${expectedLinks[index].path}`);
      expect(navLink.textContent).toEqual(expectedLinks[index].text);
    });
  });

  it('renders the logo text correctly', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    expect(screen.getByText('Math Magicians')).toBeInTheDocument();
  });
});
