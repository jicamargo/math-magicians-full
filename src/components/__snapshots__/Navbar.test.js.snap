// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the Navbar component 1`] = `
<nav
  class="navbar"
>
  <h1
    class="menu-icon"
  >
    Math Magicians
  </h1>
  <ul
    class="menu-nav"
  >
    <li>
      <a
        class=""
        href="/Home"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class=""
        href="/Calc"
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        class=""
        href="/Quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
